test/segment-loader.test.js

```diff
@@ -79,51 +79,38 @@ SegmentLoader.prototype.addSegmentMetadataCue_ = function() {};
 QUnit.module('SegmentLoader Isolated Functions');
 
 QUnit.test('getSyncSegmentCandidate works as expected', function(assert) {
-  assert.equal(getSyncSegmentCandidate(-1, {}), 0, '-1 timeline, try index 0');
-  assert.equal(getSyncSegmentCandidate(0, {}), 0, '0 timeline, no segments, try index 0');
-
-  assert.equal(
-    getSyncSegmentCandidate(0, {segments: [
-      {timeline: 0},
-      {timeline: 0}
-    ]}),
-    1,
-    '0 timeline, two timeline 0 segments, try index 1'
-  );
-
-  assert.equal(
-    getSyncSegmentCandidate(0, {segments: [
-      {timeline: 0},
-      {timeline: 0},
-      {timeline: 0},
-      {timeline: 0}
-    ]}),
-    1,
-    '0 timeline, four timeline 0 segments, try index 1'
-  );
-
-  assert.equal(
-    getSyncSegmentCandidate(0, {segments: [
-      {timeline: 0},
-      {timeline: 1},
-      {timeline: 1},
-      {timeline: 1}
-    ]}),
-    0,
-    '0 timeline, one timeline 0 segment, three timeline 1 segments, try index 0'
-  );
-
-  assert.equal(
-    getSyncSegmentCandidate(0, {segments: [
-      {timeline: 1},
-      {timeline: 1},
-      {timeline: 1},
-      {timeline: 1}
-    ]}),
-    3,
-    '0 timeline, four timeline 1 segments, try index 3'
-  );
-
+  const playlist = {};
+
+  assert.equal(getSyncSegmentCandidate(-1, playlist, 0), 0, '-1 timeline, no segments, 0 target');
+  assert.equal(getSyncSegmentCandidate(0, playlist, 0), 0, '0 timeline, no segments, 0 target');
+
+  playlist.segments = [
+    {timeline: 0, duration: 4},
+    {timeline: 0, duration: 4},
+    {timeline: 0, duration: 4},
+    {timeline: 0, duration: 4}
+  ];
+
+  assert.equal(getSyncSegmentCandidate(-1, playlist, 0), 0, '-1 timeline, 4x 0 segments, 0 target');
+  assert.equal(getSyncSegmentCandidate(0, playlist, 1), 0, '0 timeline, 4x 0 segments, 4 target');
+  assert.equal(getSyncSegmentCandidate(0, playlist, 4), 1, '0 timeline, 4x 0 segments, 4 target');
+  assert.equal(getSyncSegmentCandidate(-1, playlist, 8), 0, '-1 timeline, 4x 0 segments, 8 target');
+  assert.equal(getSyncSegmentCandidate(0, playlist, 8), 2, '0 timeline, 4x 0 segments, 8 target');
+  assert.equal(getSyncSegmentCandidate(0, playlist, 20), 3, '0 timeline, 4x 0 segments, 20 target');
+
+  playlist.segments = [
```

Comment:
I don't think this would be a valid segments array. Might be able to skip this test.

Reply:
I think it might still be a valid test for this code in isolation.